api/src/services/opportunities_v1/get_opportunity.py

```diff
@@ -31,7 +34,30 @@ def _fetch_opportunity(
 
 
 def get_opportunity(db_session: db.Session, opportunity_id: int) -> Opportunity:
-    return _fetch_opportunity(db_session, opportunity_id, load_all_opportunity_summaries=False)
+    opportunity = _fetch_opportunity(
+        db_session, opportunity_id, load_all_opportunity_summaries=False
+    )
+    opportunity_copy = copy.deepcopy(opportunity)
+
+    s3_config = S3Config()
+    s3_config.s3_endpoint_url = "http://localstack:4566"
+    s3_client = get_s3_client(
+        s3_config, boto3.Session(aws_access_key_id="NO_CREDS", aws_secret_access_key="NO_CREDS")
+    )
+
+    for opp_attachment in opportunity_copy.opportunity_attachments:
+        file_loc = opp_attachment.file_location
+        object_key = file_loc.split(f"{s3_config.s3_opportunity_bucket}/")[
+            1
+        ]  # should be same as filename in db ?
+        pre_sign_file_loc = s3_client.generate_presigned_url(
+            "get_object",
+            Params={"Bucket": s3_config.s3_opportunity_bucket, "Key": object_key},
+            ExpiresIn=1800,
```

Comment:
I'd say we define something in the s3_config class with a default of 1800 so we can override it if we want something different.